pkg/handlers/embedded_cluster_node_join_command.go

```diff
@@ -178,13 +169,41 @@ func (h *Handler) GetEmbeddedClusterNodeJoinCommand(w http.ResponseWriter, r *ht
 		return
 	}
 
-	JSON(w, http.StatusOK, GetEmbeddedClusterNodeJoinCommandResponse{
-		ClusterID:              install.Spec.ClusterID,
+	clusterUUID, err := uuid.Parse(install.Spec.ClusterID)
+	if err != nil {
+		logger.Error(fmt.Errorf("failed to parse cluster id: %w", err))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	var currentAppVersionLabel string
+	// attempt to get the current app version label from the installed app
+	installedApps, err := store.GetStore().ListInstalledApps()
+	if err == nil && len(installedApps) > 0 {
+		// "CurrentSequence" is the latest available version of the app in a non-embedded cluster.
+		// However, in an embedded cluster, the "CurrentSequence" is the latest version of the app.
+		// This is because EC uses the new upgrade flow, which only downloads and renders app releases when
+		// the app version is being installed.
```

Comment:
```suggestion
		// "CurrentSequence" is the latest available version of the app in a non-embedded cluster.
		// However, in an embedded cluster, the "CurrentSequence" is also the currently deployed version of the app.
		// This is because EC uses the new upgrade flow, which only creates a new app version when
		// the app version gets deployed. And because rollbacks are not supported in embedded cluster yet.
```